### PR TITLE
Redo of pokegame

### PR DESCRIPTION
**Pull request by appelemac**
The test fails rn because it doesn't have the dll

### CODE REVIEW

**Comment by appelemac** [comment]
I could implement one of these for security: http://stackoverflow.com/questions/299198/implement-c-sharp-generic-timeout

File: NadekoBot/Commands/Evaluate.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            
+            
+        }
+
+        class CustomParser : MathParser
+        {
+            public CustomParser() : base()
+            {
+                Console.WriteLine("Customizing here");
+                this.OperatorAction.Add("!", (x, y) => factorial(x));
```
Comment (Kwoth) [inline NadekoBot/Commands/Evaluate.cs]: this doesn't look like its doing the factorial

File: NadekoBot/Commands/Evaluate.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                this.OperatorAction.Add("_", (x, y) => 10);
```
Comment (Kwoth) [inline NadekoBot/Commands/Evaluate.cs]: whats this?

File: NadekoBot/Commands/Evaluate.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            
+            expression = Regex.Replace(expression, @"\s+", "");
+            expression = Regex.Replace(expression, @"\d+\!", new MatchEvaluator(FactorialString));
+            try
+            {
+                string result = parser.Parse(expression).ToString();
+                return result;
+            } catch(System.OverflowException e)
+            {
+                return $"Overflow error on {expression}";
```
Comment (Kwoth) [inline NadekoBot/Commands/Evaluate.cs]: Don't catch errors here, catch them in dofunc

File: NadekoBot/Commands/Evaluate.cs (hunk 2 above)
Comment (appelemac) [inline NadekoBot/Commands/Evaluate.cs]: I followed the instructions from the official site: https://blog.artemlos.net/posts/factorial-notation-in-mathos-parser-1/

**Comment by Kwoth** [comment]
checks must pass, ci is restoring the packages, anyway, ill let this one sit for a while, not necessary atm.

**Comment by appelemac** [comment]
I think the failure of the check is because the Mathos parser is not included here.

**Comment by appelemac** [comment]
I've reverted the commit for the Evaluation function, as that isn't going to be added anytime soon now.

**Comment by Kwoth** [comment]
:O When you are ready to demo this contact me on discord

File: NadekoBot/Commands/Evaluate.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            }
+            await e.Channel.SendMessage($"result: {answer}\n ticks: {sw.ElapsedTicks}\n milliseconds: {sw.ElapsedMilliseconds}");
+           
+        };
+
+        private string evaluate(string expression)
+        {
+            
+            expression = Regex.Replace(expression, @"\s+", "");
+            expression = Regex.Replace(expression, @"\d+\!", new MatchEvaluator(FactorialString));
```
Comment (tg44) [inline NadekoBot/Commands/Evaluate.cs]: Where are the FactorialString, and the MatchEvaluator? If I read well the mathosparsers doc, you can add your operator (!) to the beginning of the operator list, and it will get the biggest precedency, absolutely o need to hack it there :D

File: NadekoBot/Commands/Evaluate.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Drawing;
+using System.Threading.Tasks;
+using Discord.Commands;
+using NadekoBot.Extensions;
+using NadekoBot.Modules;
+using Mathos.Parser;
```
Comment (tg44) [inline NadekoBot/Commands/Evaluate.cs]: Where is the Nuget ref or anything for the use of this package?

File: NadekoBot/Commands/Evaluate.cs (hunk 4 above)
Comment (appelemac) [inline NadekoBot/Commands/Evaluate.cs]: aiya, I tried to follow the tutorial, but I must have been confused xD

File: NadekoBot/Commands/Evaluate.cs (hunk 5 above)
Comment (appelemac) [inline NadekoBot/Commands/Evaluate.cs]: It worked using only the Mathos.Parser...

File: NadekoBot/Commands/Evaluate.cs [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        {
+            Console.WriteLine("pass");
+            if (x == 1 || x == 0)
+            {
+               
+                return 1;
+            }
+            else
+            {
+                return x * factorial(x - 1);
```
Comment (tg44) [inline NadekoBot/Commands/Evaluate.cs]: This is regular recursion. Regular recursion is bad! Try it with loop!
http://www.codeproject.com/Articles/142292/Recursive-methods-in-Csharp

(btw its tailrec, but the jit doesn't compile it to a loop, and in theory not detecting it as a tailrec. Some langueges are autodetecting tailrec, others have annotations to make the loop conversion, sadly C# is neither of them.)

Comment (appelemac) [inline NadekoBot/Commands/Evaluate.cs]: Mesa no write, mesa copy from https://blog.artemlos.net/posts/factorial-notation-in-mathos-parser-1/

File: NadekoBot/Classes/FlowersHandler.cs [hunk 7]
```diff
@@ -13,10 +11,22 @@ internal static class FlowersHandler {
...
+                }
+                await u.SendMessage("👑Congratulations!👑\nYou got: " + flows);
+            }
+            else
+            {
+                for (var i = 0; i > amount; i--)
+                {
+                    flows += "🌸";
+                }
+                await u.SendMessage("You lost " + flows + ", poor you");
```
Comment (Kwoth) [inline NadekoBot/Classes/FlowersHandler.cs]: Make this a separate function, and probably rephrase it to "Spent".

You've spent 1 flower to heal yourself in pokemon game!

**Comment by appelemac** [comment]
The moves do need to be added from the beginning right now, maybe I'll make a command to add default moves?
